src/components/FacilitySwitcher.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+  <ion-card>
+    <ion-card-header>
+      <ion-card-title>
+        {{ "Facility" }}
+      </ion-card-title>
+    </ion-card-header>
+    <ion-card-content>
+      {{ 'Specify which facility you want to operate from. Order, inventory and other configuration data will be specific to the facility you select.' }}
+    </ion-card-content>
+    <ion-item lines="none">
+      <ion-label>{{ "Select facility" }}</ion-label>
+      <ion-select interface="popover" :value="appUserState.currentFacility.facilityId" @ionChange="setFacility($event)">
+        <ion-select-option v-for="facility in (appUserState.userProfile ? appUserState.userProfile.facilities : [])" :key="facility.facilityId" :value="facility.facilityId" >{{ facility.name }}</ion-select-option>
+      </ion-select>
+    </ion-item>
+  </ion-card>
+</template>
+
+<script setup lang="ts">
+  import { IonCard, IonCardContent, IonCardHeader, IonCardTitle, IonItem, IonLabel, IonSelect, IonSelectOption } from '@ionic/vue'
+  import { appContext } from '../index';
+  import { computed } from 'vue';
+
+  const emit = defineEmits(['facility-change-alert', 'get-facility-details' ,'update-facility-id'])
```

Comment:
Add info regarding the apps related to these events and a reference for defineEmits.